tests/phpunit/tests/db.php

```diff
@@ -2117,13 +2251,41 @@ public function data_prepare_should_respect_the_allow_unsafe_unquoted_parameters
 		);
 	}
 
+	public function test_allow_unsafe_unquoted_parameters() {
+		global $wpdb;
+
+		$sql    = 'WHERE (%i = %s) OR (%10i = %10s) OR (%5$i = %6$s)';
+		$values = array( 'field_a', 'string_a', 'field_b', 'string_b', 'field_c', 'string_c' );
+
+		$default = $wpdb->allow_unsafe_unquoted_parameters;
+
+		$property = new ReflectionProperty( $wpdb, 'allow_unsafe_unquoted_parameters' );
+		$property->setAccessible( true );
+
+		$property->setValue( $wpdb, true );
+
+		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+		$part = $wpdb->prepare( $sql, $values );
+		$this->assertSame( 'WHERE (`field_a` = \'string_a\') OR (`   field_b` =   string_b) OR (`field_c` = string_c)', $part ); // Unsafe, unquoted parameters.
+
+		$property->setValue( $wpdb, false );
+
+		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+		$part = $wpdb->prepare( $sql, $values );
+		$this->assertSame( 'WHERE (`field_a` = \'string_a\') OR (`   field_b` = \'  string_b\') OR (`field_c` = \'string_c\')', $part );
```

Comment:
Should this be a separate test, or is the preceding `$property->setValue( $wpdb, true );` and `$wpdb->prepare( $sql, $value )` required to set up the conditions for this one?

Reply:
In this case they do work together, as the `allow_unsafe_unquoted_parameters` property is `private`, and currently set to `true` (eventually it will be changed to false).

I'm using the `ReflectionProperty` so I can `setAccessible( true )` and `setValue( $wpdb, true )`, this is for the first part of the test; then I `setValue( $wpdb, false )` so I can re-run the test to check the output has changed; then finally I set it back to it's `$default` value, and `setAccessible( false )`.